api/envoy/extensions/original_ip_detection/custom_header/v3/custom_header.proto

```diff
@@ -0,0 +1,51 @@
+syntax = "proto3";
+
+package envoy.extensions.original_ip_detection.custom_header.v3;
+
+import "envoy/type/v3/http_status.proto";
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.original_ip_detection.custom_header.v3";
+option java_outer_classname = "CustomHeaderProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: Custom header original IP detection extension]
+
+// This extension allows for the original downstream remote IP to be detected
+// by reading the value from a configured header name. If the value is successfully parsed
+// as an IP, it'll be treated as the effective downstream remote address and seen as such
+// by all filters. See :ref:`original_ip_detection_extensions
+// <envoy_v3_api_field_extensions.filters.network.http_connection_manager.v3.HttpConnectionManager.original_ip_detection_extensions>`
+// for an overview of how extensions operate and what happens when an extension fails
+// to detect the remote IP.
+//
+// [#extension: envoy.http.original_ip_detection.custom_header]
+message CustomHeaderConfig {
+  // Configuration options for rejecting requests when detection fails.
+  message RejectOptions {
+    // If reject_request_if_detection_fails is set, this sets the HTTP status that is sent downstream
+    // when detection fails. The default status is HTTP 403 Forbidden.
+    type.v3.HttpStatus status_on_error = 1;
+
+    // If reject_request_if_detection_fails is set, this sets the HTTP body in the response
+    // when detection fails. If not set, an empty body is used.
```

Comment:
your call, but most envoy sendLocalReply instances do have a user-friendly string.  "envoy overloaded" or "stream timeout".  I'd lean towards a simple body but it's Ok either way.

Reply:
Now that you mention it, for our internal extension we'll probs need a DataSource... So this needs to stay configurable I am afraid...

Reply:
Oh, i'm fine with it being configurable, I just lean towards the default being non-empty. Optional though!

Reply:
Leaving it empty for now.